### DISCUSSION TITLE
Load parameter issue ?

### DISCUSSION THREAD

**Discussion by rolfdegen** [ElectroTechnique/electrotechnique.github.io#31] [Software Issues]
Hallo ElectroTechnique

Thank you for your lovely DIY synthesizer :)

I have a problem loading a patch with pulse waveform. Pulse Width Amt is always zero .

I read your code. When loading a patch first updatePWMAmount () and then follows  updatePWA () and updatePWB () . 

Into updatePWMAmouth set pwA and pwB always zero and Pulse Width in my loaded Patch is zero. 

I removed the updatePWMAmount function in  setCurrentPatchData and now it works.

Is that right !?

Your TSynth code:
` ampRelease = data[43].toFloat();
	  fxAmt = data[44].toFloat();
	  fxAmtPrevValue = fxAmt;//PICK-UP
	  fxMix = data[45].toFloat();
	  fxMixPrevValue = fxMix;//PICK-UP
	  pitchEnv = data[46].toFloat();
	  velocitySens = data[47].toFloat();
	  chordDetune = data[48].toInt();
	  //  SPARE1 = data[49].toFloat();
	  //  SPARE2 = data[50].toFloat();
	  //  SPARE3 = data[51].toFloat();

	  updatePatchname();
	  updateUnison();
	  updateWaveformA();
	  updateWaveformB();
	  updatePitchA();
	  updatePitchB();
	  updateDetune();
	  updatePWMSource();
	  updatePWMAmount();
	  updatePWA();
	  updatePWB();
	  updatePWMRate();
	  updateOscLevelA();
	  updateOscLevelB();
	  updateNoiseLevel();
	  updateFilterFreq();

FLASHMEM void updatePWMAmount() {
  //MIDI only - sets both osc
  pwA = 0;
  pwB = 0;
  setPwmMixerALFO(pwmAmtA);
  setPwmMixerBLFO(pwmAmtB);
  showCurrentParameterPage("PWM Amt", String(pwmAmtA) + " " + String(pwmAmtB));
}

FLASHMEM void updatePWA() {
  if (pwmRate == -10) {
    //if PWM amount is around zero, fixed PW is enabled
    setPwmMixerALFO(0);
    setPwmMixerBLFO(0);
    setPwmMixerAFEnv(0);
    setPwmMixerBFEnv(0);
    setPwmMixerAPW(1);
    setPwmMixerBPW(1);
    if (oscWaveformA == WAVEFORM_TRIANGLE_VARIABLE) {
      showCurrentParameterPage("1. PW Amt", pwA, VAR_TRI);
    } else {
      showCurrentParameterPage("1. PW Amt", pwA, PULSE);
    }
  } else {
    setPwmMixerAPW(0);
    setPwmMixerBPW(0);
    if (pwmSource == PWMSOURCELFO) {
	  //PW alters PWM LFO amount for waveform A
      setPwmMixerALFO(pwmAmtA);
      showCurrentParameterPage("1. PWM Amt", "LFO " + String(pwmAmtA));
    } else {
      //PW alters PWM Filter Env amount for waveform A
      setPwmMixerAFEnv(pwmAmtA);
      showCurrentParameterPage("1. PWM Amt", "F. Env " + String(pwmAmtA));
    }
  }
  float pwA_Adj = pwA;//Prevent silence when pw = +/-1.0 on pulse
  if (pwA > 0.98) pwA_Adj = 0.98f;
  if (pwA < -0.98) pwA_Adj = -0.98f;
  pwa.amplitude(pwA_Adj);
}`

Thank you for your help. Greetings from Germany. Rolf

**Comment by Electrotechnique** [comment]
Ok, I'll fix this and the save parameter bugs in the release, thanks.

**Reply by rolfdegen** [reply]
Ok. The problem was noticed when I programmed my new UI with only 4 potentiometers.  I haven't checked all the parameters yet.  Link: https://forum.pjrc.com/threads/63255-New-Teensy-4-1-DIY-Synthesizer/page6?highlight=Teensy+Shruthi
Thanks for your good work and the beautiful synthesizer :)  Greetings from germany. Rolf

**Comment by Electrotechnique** [comment]
Fixed in V2.11
